### PR TITLE
Add tests for reporting_prescriptions and fix some bugs

### PR DESCRIPTION
**Pull request by ssrihari**
**Story card:** [ch3095](https://app.shortcut.com/simpledotorg/story/3095/implement-medications-dwh-tables)

## Because

This adds tests, and fixes some bugs with reporting_prescriptions

## This addresses

- Adds tests for `reporting_prescriptions`
- Update prescriptions view to v2 to fix some bugs
- Add a rake task to refresh lookup tables
- Extract seeding lookup tables into a seed script
- Add constraints to lookup tables

### CODE REVIEW

File: db/views/reporting_prescriptions_v02.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                           AND lower(regexp_replace(raw.raw_dosage, '\s+', '', 'g')) =
+                               lower(regexp_replace(actual.dosage, '\s+', '', 'g'))
+             LEFT JOIN clean_medicine_to_dosages clean
+                       ON clean.rxcui = raw.rxcui
+             LEFT JOIN medicine_purposes purpose
+                       ON clean.medicine = purpose.name
+    WHERE patient_id = p.id
+      AND to_char(device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')), 'YYYY-MM') <= p.month_string
+      AND deleted_at is null
+      AND (is_deleted = false OR (is_deleted = true AND to_char(device_updated_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')), 'YYYY-MM') > p.month_string))
```
Comment (ssrihari) [inline db/views/reporting_prescriptions_v02.sql:42]: This boundary condition was incorrect in v1.

File: db/views/reporting_prescriptions_v02.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
+SELECT p.id as patient_id,
+       p.month_date,
+       COALESCE(sum(clean_dosage) FILTER (WHERE clean_name = 'Amlodipine'), 0)          as amlodipine,
+       COALESCE(sum(clean_dosage) FILTER (WHERE clean_name = 'Telmisartan'), 0)         as telmisartan,
+       COALESCE(sum(clean_dosage) FILTER (WHERE clean_name = 'Losartan Potassium'), 0)  as losartan,
```
Comment (ssrihari) [inline db/views/reporting_prescriptions_v02.sql:5]: This drug name was incorrect.

File: db/migrate/20211125072030_add_constraints_on_lookup_tables.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddConstraintsOnLookupTables < ActiveRecord::Migration[5.2]
+  def change
+    add_index :raw_to_clean_medicines, [:raw_name, :raw_dosage], unique: true, name: "raw_to_clean_medicines_unique_name_and_dosage"
+    add_index :clean_medicine_to_dosages, [:medicine, :dosage, :rxcui], unique: true, name: "clean_medicine_to_dosages__unique_name_and_dosage"
+    add_index :medicine_purposes, [:name], unique: true, name: "medicine_purposes_unique_name"
```
Comment (prabhanshuguptagit) [inline db/migrate/20211125072030_add_constraints_on_lookup_tables.rb:5]: Add a foreign key between `clean_medicine_to_dosages` and `medicine_purposes`.

File: lib/seed/drug_lookup_tables_seeder.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+module Seed
+  class DrugLookupTablesSeeder
```
Comment (prabhanshuguptagit) [inline lib/seed/drug_lookup_tables_seeder.rb:2]: Try `include DrugLookup`

File: spec/models/reports/prescription_spec.rb [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:four_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:four_months_ago]).telmisartan).to eq(40)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:three_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:three_months_ago]).telmisartan).to eq(40)
+
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:two_months_ago]).amlodipine).to eq(10)
+        expect(described_class.find_by(patient: patient, month_date: june_2021[:two_months_ago]).telmisartan).to eq(40)
+
+        # expect(described_class.find_by(patient: patient, month_date: june_2021[:one_month_ago]).other_bp_medications).to eq(10)
```
Comment (prabhanshuguptagit) [inline spec/models/reports/prescription_spec.rb:151]: 👾

File: spec/models/reports/prescription_spec.rb [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    dosage: dosage)
+end
+
+RSpec.describe Reports::Prescription, {type: :model, reporting_spec: true} do
+  describe "Associations" do
+    it { should belong_to(:patient) }
+  end
+
+  around do |example|
+    Timecop.freeze("June 30 2021 5:30 UTC") do
```
Comment (prabhanshuguptagit) [inline spec/models/reports/prescription_spec.rb:28]: Use the same date as in the `reporting_helper`. Comment is incorrect.

Comment (ssrihari) [reply spec/models/reports/prescription_spec.rb]: Fixed in 119553be5.

File: spec/models/reports/prescription_spec.rb (hunk 5 above)
Comment (ssrihari) [reply spec/models/reports/prescription_spec.rb]: Fixed in ea8901efcc791aea90a061273751826412afb679.

File: lib/seed/drug_lookup_tables_seeder.rb (hunk 4 above)
Comment (ssrihari) [reply lib/seed/drug_lookup_tables_seeder.rb:4]: Fixed in 3ab2b71aa9719ca105c013ab0f3991512075445f.

File: db/migrate/20211125072030_add_constraints_on_lookup_tables.rb (hunk 3 above)
Comment (ssrihari) [reply db/migrate/20211125072030_add_constraints_on_lookup_tables.rb:6]: Fixed in 3ab2b71aa9719ca105c013ab0f3991512075445f.

**Pull request merged**